tests/SprykerSdkTest/Spryk/Integration/Zed/Persistence/AddZedPersistenceRepositoryTest.php

```diff
@@ -49,9 +53,32 @@ public function testAddsZedRepositoryFileOnProjectLayer(): void
             '--mode' => 'project',
         ]);
 
-        $this->assertFileExists(
-            $this->tester->getProjectModuleDirectory()
-            . 'Persistence/FooBarRepository.php',
-        );
+        $targetClassFilePath = $this->tester->getProjectModuleDirectory() . 'Persistence/FooBarRepository.php';
+
+        $this->assertFileExists($targetClassFilePath);
+
+        $this->tester->assertClassOrInterfaceExtends($targetClassFilePath, 'Spryker\Zed\Kernel\Persistence\AbstractRepository');
+    }
+
+    /**
+     * @return void
+     */
+    public function testAddsZedRepositoryFileThatExtendsSameCoreBaseClass(): void
+    {
+        $this->tester->run($this, [
+            '--module' => 'FooBar',
+            '--mode' => 'core',
+        ]);
+
+        $this->tester->run($this, [
+            '--module' => 'FooBar',
+            '--mode' => 'project',
+        ]);
+
+        $targetClassFilePath = $this->tester->getProjectModuleDirectory('FooBar', 'Zed') . 'Persistence/FooBarRepository.php';
+
+        $this->assertFileExists($targetClassFilePath);
+
+        $this->tester->assertClassOrInterfaceExtends($targetClassFilePath, 'Spryker\Zed\FooBar\Persistence\FooBarRepository');
```

Comment:
I would expect a simpler way of checking for all cases:

´$this->tester->assertClassOrInterfaceExtends(ClassName::PROJECT_ZED_REPOSITORY, ClassName::ZED_REPOSITORY)´

It's not a blocker for me so I'll create a follow-up.